frontend/src/bundles/auth/components/auth-forms/sign-up-form/sign-up-form.tsx

```diff
@@ -35,16 +44,76 @@ const options = [
 ];
 
 const SignUpForm: React.FC<Properties> = ({ onSubmit }) => {
+    const dispatch = useAppDispatch();
+
+    const [selectedRole, setSelectedRole] = useState<ValueOf<typeof UserRole>>(
+        UserRole.TALENT,
+    );
+    const [isTermsAccepted, setIsTermsAccepted] = useState(false);
+
     const { control, errors, handleSubmit } = useAppForm<UserSignUpRequestDto>({
         defaultValues: DEFAULT_SIGN_UP_PAYLOAD,
         validationSchema: userSignUpValidationSchema,
     });
 
     const handleFormSubmit = useCallback(
         (event_: React.BaseSyntheticEvent): void => {
-            void handleSubmit(onSubmit)(event_);
+            event_.preventDefault();
+            if (selectedRole === UserRole.EMPLOYER) {
+                void handleSubmit(onSubmit)(event_);
+            } else if (isTermsAccepted) {
+                void handleSubmit(onSubmit)(event_);
+            } else {
+                const termsErrorMessage =
+                    'Please accept BSA Talents Terms to continue';
+                void dispatch(
+                    storeActions.notify({
+                        type: NotificationType.ERROR,
+                        message: termsErrorMessage,
+                    }),
+                );
+            }
+        },
+        [dispatch, handleSubmit, isTermsAccepted, onSubmit, selectedRole],
+    );
+
+    const handleRadioChange = useCallback(
+        (event: React.ChangeEvent<HTMLInputElement>) => {
+            switch (event.target.value) {
+                case UserRole.TALENT: {
+                    setSelectedRole(UserRole.TALENT);
+                    break;
+                }
+                case UserRole.EMPLOYER: {
+                    setSelectedRole(UserRole.EMPLOYER);
+                    break;
+                }
+                default: {
+                    break;
+                }
+            }
```

Comment:
```js
setSelectedRole(event.target.value)
```

Reply:
not resolved. you dont need switch case here

Reply:
replaced it like this `setSelectedRole(event.target.value === 'talent' ? UserRole.TALENT : UserRole.EMPLOYER);` because selectedRole needs valueof UserRole.